expression/builtin_json_vec.go

```diff
@@ -91,11 +94,74 @@ func (b *builtinJSONSetSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column
 }
 
 func (b *builtinJSONObjectSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinJSONObjectSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	nr := input.NumRows()
+	if len(b.args)&1 == 1 {
+		err := ErrIncorrectParameterCount.GenWithStackByArgs(ast.JSONObject)
+		return err
+	}
+
+	jsons := make([]map[string]interface{}, nr)
+	for i := 0; i < nr; i++ {
+		jsons[i] = make(map[string]interface{}, len(b.args)>>1)
+	}
+
+	argBuffers := make([]*chunk.Column, len(b.args))
+	var err error
+	for i := 0; i < len(b.args); i++ {
+		if i&1 == 0 {
+			if argBuffers[i], err = b.bufAllocator.get(types.ETString, nr); err != nil {
+				return err
+			}
+			if err = b.args[i].VecEvalString(b.ctx, input, argBuffers[i]); err != nil {
```

Comment:
We should add `defer` above this line so that when errors appear here all buffers we have allocated can be put back. 
You can add a `defer` statement like:
```
defer func(buf *Column) {
    b.bufAllocator.put(buf)
}(argBuffers[i])```

Reply:
Thanks for reviewing my code. I have fixed this problem now.